### ISSUE TITLE
Allows private property access without warnings

### ISSUE THREAD

**Issue by leanderpaul** [microsoft/TypeScript#59888]
### 🔎 Search Terms

private properties public access, private properties

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about this but have not found any answers.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=99&jsx=0&ts=3.3.3#code/MYGwhgzhAEAiCmBbA9tA3gKGtADgJwEsA3MAF3mgPMWgF5oByfYs+Bgbi2gjERxAr0AjJwC+GDMGQA7CKWgATJKnrT4AdzjKAFAEpOU2cgEA6EMgDm2pSgDaDKkgYBdXRhvITjxEA

### 💻 Code

```ts
class Demo {
  private item = 'private';
  sample = 1;
}

const demo = new Demo();
console.log(demo['item'])
demo.item
```

### 🙁 Actual behavior

I am able to access the `item` property without the typescript throwing any errors. This property is a private property so typescript is supposed to throw an error like `Property 'item' is private and only accessible within class 'Demo'.`

### 🙂 Expected behavior

The typescript should throw an error `Property 'item' is private and only accessible within class 'Demo'.`

**Comment by MartinJohns** [comment]
This is working as intended. Duplicate of #26328.
